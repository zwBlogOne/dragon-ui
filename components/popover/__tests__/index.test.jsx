import React from 'react';
import { render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Popover from '../index';
import Button from '../../button';

describe('Popover', () => {
  it('renders normal Popover correctly', () => {
    const wrapper = render(
      <div>
        <Popover content="确认删除吗" direction="top">
          <Button>Delete</Button>
        </Popover>
      </div>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Popover correctly with customized props', () => {
    const wrapper = render(
      <div>
        <Popover content="确认删除吗" direction="top" radius className="customized">
          <Button>Delete</Button>
        </Popover>
      </div>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Popover correctly with render props', () => {
    const wrapper = render(
      <div>
        <Popover content={() => '确认删除吗?'} direction="top">
          <Button>Delete</Button>
        </Popover>
      </div>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Popover correctly with mask', () => {
    const wrapper = render(
      <div>
        <Popover mask content={() => '确认删除吗?'} direction="top">
          <Button>Delete</Button>
        </Popover>
      </div>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('behaves correctly when clicking on mask', () => {
    const onMaskClick = jest.fn();
    const wrapper = render(
      <div>
        <Popover
          mask
          onMaskClick={onMaskClick}
          content={() => '确认删除吗?'}
          direction="top"
        >
          <Button>Delete</Button>
        </Popover>
      </div>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('behaves correctly when toggling visible status', () => {
    const wrapper = mount(
      <Popover
        mask
        content={() => '确认删除吗?'}
        direction="top"
      >
        <Button>Delete</Button>
      </Popover>
    );

    expect(wrapper.find('.ui-popover-content').hasClass('ui-popover-content-show')).toBeFalsy();
    wrapper.setProps({ visible: true });
    expect(wrapper.find('.ui-popover-content').hasClass('ui-popover-content-show')).toBeTruthy();
  });
});

