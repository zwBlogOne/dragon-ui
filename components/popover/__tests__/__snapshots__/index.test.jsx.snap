// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover behaves correctly when clicking on mask 1`] = `
<div>
  <div
    class="ui-popover"
  >
    <div
      class="ui-popover-mask"
    />
    <div
      class="ui-popover-content ui-popover-content-radius"
    >
      确认删除吗?
      <span
        class="ui-popover-arrow"
      />
    </div>
    <button
      class="ui-button theme-default"
      type="button"
    >
      Delete
    </button>
  </div>
</div>
`;

exports[`Popover renders Popover correctly with customized props 1`] = `
<div>
  <div
    class="ui-popover customized"
  >
    <div
      class="ui-popover-content ui-popover-content-radius"
    >
      确认删除吗
      <span
        class="ui-popover-arrow"
      />
    </div>
    <button
      class="ui-button theme-default"
      type="button"
    >
      Delete
    </button>
  </div>
</div>
`;

exports[`Popover renders Popover correctly with mask 1`] = `
<div>
  <div
    class="ui-popover"
  >
    <div
      class="ui-popover-mask"
    />
    <div
      class="ui-popover-content ui-popover-content-radius"
    >
      确认删除吗?
      <span
        class="ui-popover-arrow"
      />
    </div>
    <button
      class="ui-button theme-default"
      type="button"
    >
      Delete
    </button>
  </div>
</div>
`;

exports[`Popover renders Popover correctly with render props 1`] = `
<div>
  <div
    class="ui-popover"
  >
    <div
      class="ui-popover-content ui-popover-content-radius"
    >
      确认删除吗?
      <span
        class="ui-popover-arrow"
      />
    </div>
    <button
      class="ui-button theme-default"
      type="button"
    >
      Delete
    </button>
  </div>
</div>
`;

exports[`Popover renders normal Popover correctly 1`] = `
<div>
  <div
    class="ui-popover"
  >
    <div
      class="ui-popover-content ui-popover-content-radius"
    >
      确认删除吗
      <span
        class="ui-popover-arrow"
      />
    </div>
    <button
      class="ui-button theme-default"
      type="button"
    >
      Delete
    </button>
  </div>
</div>
`;
